<html><h3>:x: ci:test - 3 out of 6 jobs passed in 58 minutes 47 seconds 946 ms</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/com-liferay-commerce/tree/7.1.x">7.1.x</a><br/>Branch GIT ID: <a href="https://github.com/liferay/com-liferay-commerce/commit/0189ab75a2894a7a113d4f7f1c4607a63059b129">0189ab75a2894a7a113d4f7f1c4607a63059b129</a></p>3 out of 6 jobs PASSED<div><h4>3 Failed Jobs:</h4><ul><li><a href="https://test-1-2.liferay.com/job/test-subrepository-acceptance-pullrequest(7.1.x)/3/"><strike><strong>test-subrepository-acceptance-pullrequest(7.1.x)</strong></strike></a></li><li><a href="https://test-1-10.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/4/"><strike><strong>test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-assemble-jdk8</strong></strike></a></li><li><a href="https://test-1-11.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/1/"><strike><strong>test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-integration-mysql56-jdk8</strong></strike></a></li></ul></div><details><summary><strong>3 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-8.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/1/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/functional-smoke-tomcat80-mysql56-jdk8/0</a></li><li><a href="https://test-1-21.liferay.com/job/test-subrepository-acceptance-pullrequest-dist(7.1.x)/3/">test-subrepository-acceptance-pullrequest-dist(7.1.x)/jdk8</a></li><li><a href="https://test-1-2.liferay.com/job/test-subrepository-acceptance-pullrequest-validation(7.1.x)/3/">test-subrepository-acceptance-pullrequest-validation(7.1.x)/jdk8</a></li></ul></details><h5>For more details click <a href="https://test-1-2.liferay.com/userContent/jobs/test-subrepository-acceptance-pullrequest(7.1.x)/builds/3/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><h5><a href="https://test-1-10.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/4/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-assemble-jdk8</a></h5><div><h6>Job Results:</h6><p>0 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-10.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/AXIS_VARIABLE=0,label_exp=!master/4//consoleText">AXIS_VARIABLE=0,label_exp=!master #4</a><pre><code>     [echo] 	at org.apache.tools.ant.taskdefs.optional.XMLValidateTask.doValidate(XMLValidateTask.java:574)
     [echo] 	at org.apache.tools.ant.taskdefs.optional.XMLValidateTask.execute(XMLValidateTask.java:321)
     [echo] 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:293)
     [echo] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
     [echo] 	at org.gradle.api.internal.project.ant.BasicAntBuilder.nodeCompleted(BasicAntBuilder.java:78)
     [echo] 	at org.gradle.api.internal.project.ant.BasicAntBuilder.doInvokeMethod(BasicAntBuilder.java:103)
     [echo] 	at com.liferay.gradle.plugins.tlddoc.builder.tasks.ValidateSchemaTask.validateSchema(ValidateSchemaTask.java:123)
     [echo] 	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
     [echo] 	at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.doExecute(DefaultTaskClassInfoStore.java:141)
     [echo] 	at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:134)
     [echo] 	at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:123)
     [echo] 	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:632)
     [echo] 	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:615)
     [echo] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:95)
     [echo] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:76)
     [echo] 	... 62 more
     [echo] 
     [echo] The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/build-common.xml:845: /opt/dev/projects/github/liferay-portal-7.1.x/build-common.xml:847: exec returned: 1
      [get] Getting: http://test-1-10/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/AXIS_VARIABLE=0,label_exp=!master/4//consoleText
      [get] To: /opt/dev/projects/github/liferay-portal-7.1.x/20180814175930144.txt
   [delete] Deleting: /opt/dev/projects/github/liferay-portal-7.1.x/20180814175930144.txt
   [delete] Deleting: /opt/dev/projects/github/liferay-portal-7.1.x/null89840144.properties</code></pre></div></li></ol></div></li><li><div><h5><a href="https://test-1-11.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/1/">test-subrepository-acceptance-pullrequest-batch(7.1.x)/subrepository-integration-mysql56-jdk8</a></h5><div><h6>Job Results:</h6><p>0 Tests Passed.<br/>1 Test Failed.</p></div><ol><li><div><a href="https://test-1-11.liferay.com/job/test-subrepository-acceptance-pullrequest-batch(7.1.x)/AXIS_VARIABLE=0,label_exp=!master/1//consoleText">AXIS_VARIABLE=0,label_exp=!master #1</a><pre><code>     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/build-test-batch.xml:2080: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/build-test-batch.xml:2263: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/build-test.xml:9160: The following error occurred while executing this line:
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/build-test.xml:2415: Sourced file: inline evaluation of: `` 				 					import com.liferay.jenkins.results.parser.Job; 					import com.lifera . . . '' : Typed variable declaration : Method Invocation TestClassGroupFactory.newBatchTestClassGroup : at Line: 11 : in file: inline evaluation of: `` 				 					import com.liferay.jenkins.results.parser.Job; 					import com.lifera . . . '' : TestClassGroupFactory .newBatchTestClassGroup ( project .getProperty ( "test.batch.name" ) , job ) 
     [echo] 
     [echo] Target exception: java.lang.IllegalArgumentException: Unknown test class group

BUILD FAILED
/opt/dev/projects/github/liferay-jenkins-ee/commands/build-test-subrepository-acceptance-pullrequest-batch.xml:303: The following error occurred while executing this line:
/opt/dev/projects/github/liferay-portal-7.1.x/build-test-batch.xml:4859: The following error occurred while executing this line:
/opt/dev/projects/github/liferay-portal-7.1.x/build-test-batch.xml:2080: The following error occurred while executing this line:
/opt/dev/projects/github/liferay-portal-7.1.x/build-test-batch.xml:2263: The following error occurred while executing this line:
/opt/dev/projects/github/liferay-portal-7.1.x/build-test.xml:9160: The following error occurred while executing this line:
/opt/dev/projects/github/liferay-portal-7.1.x/build-test.xml:2415: Sourced file: inline evaluation of: `` 				 					import com.liferay.jenkins.results.parser.Job; 					import com.lifera . . . '' : Typed variable declaration : Method Invocation TestClassGroupFactory.newBatchTestClassGroup : at Line: 11 : in file: inline evaluation of: `` 				 					import com.liferay.jenkins.results.parser.Job; 					import com.lifera . . . '' : TestClassGroupFactory .newBatchTestClassGroup ( project .getProperty ( "test.batch.name" ) , job ) 

Target exception: java.lang.IllegalArgumentException: Unknown test class group</code></pre></div></li></ol></div></li></ol></details></html>

--- a/commerce-payment-engine-paypal/src/main/java/com/liferay/commerce/payment/engine/paypal/internal/PayPalCommercePaymentEngine.java
+++ b/commerce-payment-engine-paypal/src/main/java/com/liferay/commerce/payment/engine/paypal/internal/PayPalCommercePaymentEngine.java
@@ -15,7 +15,6 @@
 package com.liferay.commerce.payment.engine.paypal.internal;
 
 import com.liferay.commerce.currency.model.CommerceCurrency;
-import com.liferay.commerce.currency.service.CommerceCurrencyLocalService;
 import com.liferay.commerce.exception.CommercePaymentEngineException;
 import com.liferay.commerce.model.CommerceAddress;
 import com.liferay.commerce.model.CommerceCountry;
@@ -376,14 +375,6 @@ public class PayPalCommercePaymentEngine implements CommercePaymentEngine {
 			ServiceContext serviceContext)
 		throws Exception {
 
-		CommerceCurrency commerceCurrency =
-			_commerceCurrencyLocalService.fetchPrimaryCommerceCurrency(
-				commerceOrder.getSiteGroupId());
-
-		if (commerceCurrency == null) {
-			throw new CommercePaymentEngineException.MustSetPrimaryCurrency();
-		}
-
 		APIContext apiContext = _getAPIContext(commerceOrder);
 
 		Payment payment = new Payment();
@@ -405,7 +396,8 @@ public class PayPalCommercePaymentEngine implements CommercePaymentEngine {
 
 		payment.setTransactions(
 			_getTransactions(
-				commerceOrder, commerceCurrency, serviceContext.getLocale()));
+				commerceOrder, commerceOrder.getCommerceCurrency(),
+				serviceContext.getLocale()));
 
 		payment = payment.create(apiContext);
 
@@ -432,9 +424,6 @@ public class PayPalCommercePaymentEngine implements CommercePaymentEngine {
 
 	private static final DecimalFormat _PAY_PAL_DECIMAL_FORMAT =
 		new DecimalFormat("#,###.00");
-
-	@Reference
-	private CommerceCurrencyLocalService _commerceCurrencyLocalService;
 
 	@Reference
 	private ConfigurationProvider _configurationProvider;

--- a/commerce-payment-engine-paypal/src/main/java/com/liferay/commerce/payment/engine/paypal/internal/PayPalCommercePaymentEngine.java
+++ b/commerce-payment-engine-paypal/src/main/java/com/liferay/commerce/payment/engine/paypal/internal/PayPalCommercePaymentEngine.java
@@ -62,6 +62,8 @@ import com.paypal.api.payments.ShippingAddress;
 import com.paypal.api.payments.Transaction;
 import com.paypal.base.rest.APIContext;
 
+import java.text.DecimalFormat;
+
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
@@ -238,16 +240,14 @@ public class PayPalCommercePaymentEngine implements CommercePaymentEngine {
 		Details details = new Details();
 
 		details.setShipping(
-			String.valueOf(
-				commerceCurrency.round(commerceOrder.getShippingAmount())));
+			_PAY_PAL_DECIMAL_FORMAT.format(commerceOrder.getShippingAmount()));
 		details.setSubtotal(
-			String.valueOf(
-				commerceCurrency.round(commerceOrder.getSubtotal())));
+			_PAY_PAL_DECIMAL_FORMAT.format(commerceOrder.getSubtotal()));
 
 		amount.setDetails(details);
 
 		amount.setTotal(
-			String.valueOf(commerceCurrency.round(commerceOrder.getTotal())));
+			_PAY_PAL_DECIMAL_FORMAT.format(commerceOrder.getTotal()));
 
 		return amount;
 	}
@@ -316,8 +316,8 @@ public class PayPalCommercePaymentEngine implements CommercePaymentEngine {
 			item.setDescription(cpDefinition.getShortDescription(languageId));
 			item.setName(commerceOrderItem.getName(languageId));
 			item.setPrice(
-				String.valueOf(
-					commerceCurrency.round(commerceOrderItem.getUnitPrice())));
+				_PAY_PAL_DECIMAL_FORMAT.format(
+					commerceOrderItem.getUnitPrice()));
 			item.setQuantity(String.valueOf(commerceOrderItem.getQuantity()));
 			item.setSku(commerceOrderItem.getSku());
 
@@ -429,6 +429,9 @@ public class PayPalCommercePaymentEngine implements CommercePaymentEngine {
 		return new CommercePaymentEngineResult.StartPayment(
 			payment.toJSON(), url);
 	}
+
+	private static final DecimalFormat _PAY_PAL_DECIMAL_FORMAT =
+		new DecimalFormat("#,###.00");
 
 	@Reference
 	private CommerceCurrencyLocalService _commerceCurrencyLocalService;

